command.rb

```diff
@@ -88,10 +133,73 @@ def format_log(severity, timestamp, progname, msg)
       color = '1;30'
     end
 
+    str = ''
+
     if @options[:use_color]
-      "\e[#{color}m#{prefix}, [#{time}] : #{msg}\e[0m\n"
+      str += "\e[#{color}m#{prefix}, [#{time}] : #{msg}\e[0m"
     else
-      "#{prefix}, [#{time}] : #{msg}\n"
+      str += "#{prefix}, [#{time}] : #{msg}"
+    end
+
+    str
+  end
+
+
+  def format_log(severity, timestamp, progname, msg)
+    str = format_msg(severity, timestamp, progname, msg)
+
+    if @options[:use_color]
+      str = "\e[F#{str}\e[K" if @status_printed
+    else
+      str = "\n#{str}" if @status_printed
+    end
+
+    @status_printed = false
+
+    str += "\n"
+
+    str
+  end
+
+  def symbolize_keys!(hash)
+    hash.keys.each do |key|
+      hash[(key.to_sym rescue key) || key] = hash[key]
+    end
+  end
+
+  def each_with_status_and_delay(things, template: '%{id}...', &blk)
+    things.each_with_index do |thing, idx|
+      symbolize_keys!(thing)
+      str = "#{idx+1}/#{things.length}: "
+      str += (template % thing)
+      status(str)
+      blk.call(thing)
+    end
+  end
+
+  def with_retries
+    retries = @options[:retries]
+    status_code = nil
+    tries = 0
+    begin
+      sleep(@options[:delay] + tries)
+      tries += 1
+      status_code, definition = yield
+      raise RetryError.new("Got status_code #{status_code}") if status_code != '200'
+      return [status_code, definition]
+    rescue RetryError, Timeout::Error, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+           Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError => e
+
+      if tries > retries
+        @logger.error("Failed (out of retries)")
+        return [status_code || -1, nil]
+      end
+
+      keeping_status do
+        @logger.warn("Received error: (#{e.class}) #{e.message}, retrying (#{tries}/#{retries})...")
+      end
+
+      retry
```

Comment:
Yeah, I ran into it incidentally. There's more than just that, it's kinda crazy!